lib/endpoints/class-wp-rest-posts-terms-controller.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+class WP_REST_Posts_Terms_Controller extends WP_REST_Controller {
+
+	protected $post_type;
+
+	public function __construct( $post_type ) {
+		$this->post_type = $post_type;
+		$this->posts_controller = new WP_REST_Posts_Controller( $post_type );
+	}
+
+	/**
+	 * Register the routes for the objects of the controller.
+	 */
+	public function register_routes() {
+
+		$base = $this->posts_controller->get_post_type_base( $this->post_type );
+
+		register_rest_route( 'wp/v2', '/' . $base . '/(?P<id>[\d]+)/terms/(?P<taxonomy>[^/]+)', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+		) );
+
+		register_rest_route( 'wp/v2', '/' . $base . '/(?P<id>[\d]+)/terms/(?P<taxonomy>[^/]+)/(?P<term_id>[\d]+)', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_item' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+			array(
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'create_item' ),
+				'permission_callback' => array( $this, 'create_item_permissions_check' ),
+			),
+			array(
+				'methods'         => WP_REST_Server::DELETABLE,
+				'callback'        => array( $this, 'delete_item' ),
+				'permission_callback' => array( $this, 'create_item_permissions_check' ),
+			),
+		) );
+	}
+
+	/**
+	 * Get a collection of posts
+	 *
+	 * @param WP_REST_Request $request Full details about the request
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
```

Comment:
Not 100% sure. I guess for consistency we should add it? I do wonder whether typically developers would expect this or not. Pagination on object propper is somewhat expected, but how many people doing a simple GET that returns 10 tags would think to look for more?

This may be made a bit better with the wrapper object for collection responses to make it more obvious there is more results available.